### PR TITLE
build: stricter unused variable checking

### PR DESCRIPTION
**Pull request by DevVersion**
Currently only variables that don't start with an underscore will be checked by TSLint.

This doesn't make sense, because variables get prefixed with an underscore to indicate that they are not part of the public API, but still need to be public in favor of AOT compilation for example. This means that those properties are having the public modifier set (explicitly or automatically) and therefore won't be treated as unused anyway.

**TL;DR**: With this change, prefixed variables will be checked by the `no-unused-variable` TSLint rule and those which are marked as `private` and are unused will be catched by TSLint.

### CODE REVIEW

File: src/lib/paginator/paginator.spec.ts [hunk 1]
```diff
@@ -281,7 +281,7 @@ class MdPaginatorApp {
 
   @ViewChild(MdPaginator) mdPaginator: MdPaginator;
 
-  constructor(private _elementRef: ElementRef) { }
+  constructor() {}
```
Comment (crisbeto) [inline src/lib/paginator/paginator.spec.ts]: You can remove this constructor.

Comment (DevVersion) [reply src/lib/paginator/paginator.spec.ts]: Done.

File: src/cdk/rxjs/rx-operators.ts [hunk 2]
```diff
@@ -89,6 +89,7 @@ export class StartWithBrand { private _; }
 export class DebounceTimeBrand { private _; }
 export class AuditTimeBrand { private _; }
 export class TakeUntilBrand { private _; }
```
Comment (crisbeto) [inline src/cdk/rxjs/rx-operators.ts]: I'd be ok with having these properties be public. They're not a part of any API or generated JS anyway.

Comment (DevVersion) [reply src/cdk/rxjs/rx-operators.ts]: Yeah I don't mind either. I just changed it. (doesn't really matter)

Comment (jelbourn) [reply src/cdk/rxjs/rx-operators.ts]: The brand properties actually _must_ be `private`, otherwise they are structurally identical to anything and everything.
